keel-artifact/src/test/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListenerTests.kt

```diff
@@ -25,36 +27,40 @@ import io.mockk.verify
 import org.springframework.context.ApplicationEventPublisher
 
 internal class ArtifactListenerTests : JUnit5Minutests {
-  val korkDeb = Artifact.builder()
-    .type("DEB")
-    .customKind(false)
-    .name("fnord")
-    .version("0.156.0-h58.f67fe09")
-    .reference("debian-local:pool/f/fnord/fnord_0.156.0-h58.f67fe09_all.deb")
-    .metadata(mapOf("releaseStatus" to FINAL))
-    .provenance("https://my.jenkins.master/jobs/fnord-release/58")
-    .build()
+  val publishedDeb = PublishedArtifact(
+    type = "DEB",
+    customKind = false,
+    name = "fnord",
+    version = "0.156.0-h58.f67fe09",
+    reference = "debian-local:pool/f/fnord/fnord_0.156.0-h58.f67fe09_all.deb",
+    metadata = mapOf("releaseStatus" to FINAL),
+    provenance = "https://my.jenkins.master/jobs/fnord-release/58"
+  )
 
   val debianArtifact = DebianArtifact(name = "fnord", deliveryConfigName = "fnord-config", vmOptions = VirtualMachineOptions(baseOs = "bionic", regions = setOf("us-west-2")))
   val dockerArtifact = DockerArtifact(name = "fnord/myimage", tagVersionStrategy = BRANCH_JOB_COMMIT_BY_JOB, deliveryConfigName = "fnord-config")
   val deliveryConfig = DeliveryConfig(name = "fnord-config", application = "fnord", serviceAccount = "keel", artifacts = setOf(debianArtifact, dockerArtifact))
 
   data class ArtifactFixture(
-    val event: KorkArtifactEvent,
+    val event: ArtifactPublishedEvent,
     val artifact: DeliveryArtifact,
     val repository: KeelRepository = mockk(relaxUnitFun = true),
     val artifactService: ArtifactService = mockk(relaxUnitFun = true),
     val clouddriverService: CloudDriverService = mockk(relaxUnitFun = true),
     val publisher: ApplicationEventPublisher = mockk(relaxUnitFun = true)
   ) {
-    val listener: ArtifactListener = ArtifactListener(repository, artifactService, clouddriverService, publisher)
+    private val eventBridge = SpringEventPublisherBridge(publisher)
```

Comment:
What is this eventBridge?

Reply:
This is needed because we cannot have a dependency on any Spring libraries in `keel-api`, since it's the library that plugin implementors will use (and it also causes problems with the PF4J stuff, I think -- @robzienert would know more). So, the solution Rob came up with, which I just replicated for artifact plugins, was to define a new interface called `EventPublisher` that looks-and-feels like Spring's `ApplicationEventPublisher`, which we can use in `keel-api` and plugins specifically. In `keel`, we just wire in this bean which  is a "bridge" between the generic interface and the Spring implementation.

Reply:
Ah, interesting! Thanks.